### DISCUSSION TITLE
Getting wrong "is a required field" for lists

### DISCUSSION THREAD

**Discussion by MoSattler** [edmundhung/conform#626] [Q&A]
### Describe the bug and the expected behavior

In certain situations, when using lists, default values and `useInputControl`, I get falsely flagged "is required field" errors.

### Conform version

1.1.0

### Steps to Reproduce the Bug or Issue

```tsx
import { submit } from './actions'
import { useFormState } from 'react-dom'
import { parseWithYup } from '@conform-to/yup'
import {
  useForm,
  getFormProps,
  useInputControl,
  FieldMetadata
} from '@conform-to/react'
import { object, string, tuple } from 'yup'

export const formSchema = object({
  myTupleOfStrings: tuple([
    string().required(), 
    string().required()
  ]).required()
})

const CustomComponenWrapper = ({ field }: { field: FieldMetadata<string> }) => {
  const originalFileNameFieldControl = useInputControl(field)

  return (
    <CustomComponent
      value={originalFileNameFieldControl.value}
      onChange={(value) => originalFileNameFieldControl.change(value)}
    />
  )
}

function NewModuleRunPage() {
  const [lastResult, action] = useFormState(submit, undefined)
  const [form, fields] = useForm({
    lastResult,
    defaultValue: {
      myTupleOfStrings: [null, null]
    },
    onValidate({ formData }) {
      return parseWithYup(formData, { schema: formSchema })
    }
  })

  return (
    <form action={action} {...getFormProps(form)}>
      {fields.myTupleOfStrings.getFieldList().map((field) => (
        <CustomComponenWrapper key={field.id} field={field} />
      ))}
      <button type="submit">Submit</button>
    </form>
  )
}
```

### Additional context

In the example, if I submit right away, without changing any values, I get:

```
myTupleOfStrings is a required field
```
even though `myTupleOfStrings` is defined in the default values.

 I'd expect a message 
```
myTupleOfStrings["0"] is a required field
```
and I indeed get that if only one of the two values is set to `null`.

**Comment by edmundhung** [comment]
@MoSattler Can you prepare a repo so I can debug it further? I guess this is a bug related to how `useInputControl` works

**Comment by MoSattler** [comment]
@edmundhung Will a [code sandbox](https://codesandbox.io/p/devbox/nameless-fast-6ysvwg?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clvds61he00053b5vpl39ru5k%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clvds61he00023b5vysgi7c5g%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clvds61he00033b5vj8mo63gk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clvds61he00043b5vr1uxkc8l%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clvds61he00023b5vysgi7c5g%2522%253A%257B%2522id%2522%253A%2522clvds61he00023b5vysgi7c5g%2522%252C%2522activeTabId%2522%253A%2522clvds87bl00ac3b5vjclk6tkx%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clvds61hd00013b5v4de4aq01%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252Ftasks.json%2522%252C%2522id%2522%253A%2522clvds87bl00ac3b5vjclk6tkx%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clvds61he00043b5vr1uxkc8l%2522%253A%257B%2522id%2522%253A%2522clvds61he00043b5vr1uxkc8l%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clvds7z48007r3b5vhbpggw6t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clvds7z48007r3b5vhbpggw6t%2522%257D%252C%2522clvds61he00033b5vj8mo63gk%2522%253A%257B%2522id%2522%253A%2522clvds61he00033b5vj8mo63gk%2522%252C%2522activeTabId%2522%253A%2522clvds7tu4006p3b5vw92kikf8%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clvds7tu4006p3b5vw92kikf8%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) sufice?

**Comment by edmundhung** [comment]
Thanks for the repo. The issue is indeed related to `useInputControl` in which it doesn't render an hidden input for you by default and so Conform treats it as an empty array instead of an array of two items with empty value. If you render an hidden input inside the `<CustomComponenWrapper />`, the `useInputControl` hook will pick it up and validate just fine.

Unfortunately I don't think I would be able to fix the issue in `useInputControl` directly because of the complexity on managing an hidden input for all kind of usecase. I would recommend giving the experimetnal `useControl` hook introduced in [v1.0.2](https://github.com/edmundhung/conform/releases/tag/v1.0.2) a try.
